src/Umbraco.Web/PropertyEditors/MultiUrlPickerPropertyEditor.cs

```diff
@@ -0,0 +1,264 @@
+                                }
+                                link.Published = false;
+                                link.Trashed = true;
+                                link.Url = deletedLocalization;
+                            }
+                            else
+                            {
+                                string entityType =
+                                    Equals(entity.AdditionalData["NodeObjectTypeId"], Constants.ObjectTypes.MediaGuid)
+                                        ? Constants.UdiEntityType.Media
+                                        : Constants.UdiEntityType.Document;
+
+                                var udi = new GuidUdi(entityType, entity.Key);
+
+                                string contentTypeAlias = (string) entity.AdditionalData["ContentTypeAlias"];;
+                                if (entity.Trashed)
+                                {
+                                    if (recycleBinLocalization == null)
+                                    {
+                                        recycleBinLocalization = services.TextService.Localize("general/recycleBin");
+
+                                    }
+                                    link.Trashed = true;
+                                    link.Url = recycleBinLocalization;
+                                }
+
+                                if (udi.EntityType == Constants.UdiEntityType.Document)
+                                {
+                                    IContentType contentType = contentTypeService.GetContentType(contentTypeAlias);
+
+                                    if (contentType == null)
+                                    {
+                                        continue;
+                                    }
+
+                                    link.Icon = contentType.Icon;
+                                    link.Published = Equals(entity.AdditionalData["IsPublished"], true);
+
+                                    if (link.Trashed == false)
+                                    {
+                                        link.Url = umbHelper.Url(entity.Id, UrlProviderMode.Relative);
+                                    }
+                                }
+                                else
+                                {
+                                    link.IsMedia = true;
+
+                                    IMediaType mediaType = contentTypeService.GetMediaType(contentTypeAlias);
+
+                                    if (mediaType == null)
+                                    {
+                                        continue;
+                                    }
+
+                                    link.Icon = mediaType.Icon;
+
+                                    if (link.Trashed == false)
+                                    {
+                                        link.Url = umbHelper.TypedMedia(entity.Id).Url;
```

Comment:
Should this line has a null check? E.g.
```
var media = umbHelper.TypedMedia(entity.Id);
if (media != null) 
{
     link.Url = media.Url;
}
```

Reply:
Shouldn't be necessary because of the entity null check earlier, but I've added it anyways since the media index/cache could be corrupted or return null for some reason